design-documents/frontend/server-side-rendering.md

```diff
@@ -0,0 +1,129 @@
+# Server-Side Rendering (SSR)
+
+_This document refers to SSR in the context of a React web application (e.g., PWA). The concept of SSR does not apply to native apps._
+
+_SSR belongs to the BFF layer of [service isolation](https://github.com/magento/architecture/tree/master/design-documents/service-isolation)._
+
+SSR can be a challenge to implement correctly, but the benefits have the potential to provide the unparalleled performance you seek to decrease bounce rate and drive more business.
+
+## Pros
+
+- Faster, first meaningful paint.
+- No need to render initial content via JavaScript on the client.
+- SEO/bot support.
+- Better SEO performance.
+- JavaScript code can be shared universally (full-stack).
+
+## Cons
+
+- User may see some content before it's interactive.
+- Difficult or tedious to implement.
+- May require additional configuration and refactoring.
+- Some features [not yet supported](#not-supported-yet).
+
+## How it works
+
+The basic idea is that we don't want the web browser (client) to be responsible for rendering content via JavaScript, because JavaScript is notoriously slow, especially at rendering DOM and especially on older or mobile devices. Unnecessary rendering also consumes more energy, which has the potential to suck a mobile device's battery dry. :battery:
```

Comment:
> we don't want the web browser (client) to be responsible for rendering content via JavaScript

Probably would be more clear to say

> we don't want the web browser (client) to be responsible for rendering the initial page's content via JavaScript

The first statement seems not-ideal because:

1. There will still be JS-based rendering (just not for the first bits of DOM)
2. SSR won't kick in for subsequent page navs in the same tab (history API + React client-side)